runtime/parser2/declaration.go

```diff
@@ -157,32 +164,60 @@ func parseParameterList(p *parser) (parameterList *ast.ParameterList) {
 	p.skipSpaceAndComments(true)
 
 	if !p.current.Is(lexer.TokenParenOpen) {
-		panic(fmt.Errorf("expected '(' as start of parameter list, got %s", p.current.Type))
+		panic(fmt.Errorf(
+			"expected %q as start of parameter list, got %q",
+			lexer.TokenParenOpen,
+			p.current.Type,
+		))
 	}
 
 	startPos := p.current.StartPos
 	p.next()
 
 	var endPos ast.Position
 
+	expectParameter := true
+
 	atEnd := false
 	for !atEnd {
 		p.skipSpaceAndComments(true)
 		switch p.current.Type {
```

Comment:
We will also need to implement `lexer.TokenUnderscore` to handle un-labeled parameters.

Reply:
Handling which will likely live in `parseParameter` I assume?

Reply:
We don't actually need to differentiate between underscore `_` and other identifiers on the lexer level. Identifiers may start with an underscore and may only contain one character, so it already covers the argument label case. See https://github.com/onflow/cadence/blob/505dc9b920954018e75410c4d26c0218bf9e0bc4/runtime/parser2/lexer/state.go#L91-L92

Reply:
Cool wasn't sure if `_` was considered an identifier -- looks like it is 👍